src/util/RandHasher.cpp

```diff
@@ -0,0 +1,32 @@
+// Copyright 2021 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "RandHasher.h"
+#include "Math.h"
+#include "fmt/format.h"
+#include <mutex>
+
+namespace stellar
+{
+namespace randHash
+{
+
+size_t gMixer{0};
+bool gHaveInitialized{false};
+static std::mutex gInitMutex;
+
+void
+initialize()
+{
+    std::lock_guard<std::mutex> guard(gInitMutex);
+    if (!gHaveInitialized)
```

Comment:
you could also do what was done in the header (use a static "local" variable: this is perfectly defined behavior and thread safe per standard)

Reply:
No, in this case we explicitly want to differentiate the case of "user initialized this variable by calling `initialize` from a known position in the code" (which means it'll always have the same relationship to gRandomEngine's seed) and "variable was initialized whenever it was first used" (which may change the point in gRandomEngine's trajectory it initializes from, depending on workload, eg. we can get a different initializer if we run with `[tx]` than we get if we run with `someSpecificTest`).

So we do not want a compiler-generated initialization flag, we want one we manually track so that we can treat failure-to-set-it as an error.